control/pid.go

```diff
@@ -22,25 +22,51 @@ func (l *Loop) newPID(config BlockConfig, logger logging.Logger) (Block, error)
 
 // BasicPID is the standard implementation of a PID controller.
 type basicPID struct {
-	mu       sync.Mutex
-	cfg      BlockConfig
-	error    float64
-	kI       float64
-	kD       float64
-	kP       float64
-	int      float64
+	mu     sync.Mutex
+	cfg    BlockConfig
+	logger logging.Logger
+	// used by the single input/output controller
+	error float64
+	kI    float64
+	kD    float64
+	kP    float64
+	int   float64
+	tuner pidTuner
+
+	// MIMO gains + state
+	PIDSets  []*PIDConfig
+	tuners   []*pidTuner
+	useMulti bool
+
+	// used by both
 	y        []*Signal
 	satLimUp float64 `default:"255.0"`
 	limUp    float64 `default:"255.0"`
 	satLimLo float64
 	limLo    float64
-	tuner    pidTuner
-	tuning   bool
-	logger   logging.Logger
 }
 
+// GetTuning returns whether the PID block is currently tuning any signals.
 func (p *basicPID) GetTuning() bool {
-	return p.tuning
+	// using locks so we do not check for tuning mid reconfigure or mid tune
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	return p.getTuning()
+}
+
+func (p *basicPID) getTuning() bool {
+	multiTune := false
+
+	if p.useMulti {
+		for _, tuner := range p.tuners {
+			// the tuners  for MIMO only get created if we want to tune
```

Comment:
this might become clear later on in my review, but if we only create tuners for PID blocks that need to be tuned, will there be any issues with the tuners being associated with the wrong blocks in a list or anything like that?